### PR TITLE
open rocksdb graph store using multi-threads when multi disks

### PR DESCRIPTION
**Pull request by zhoney**
Change-Id: I83525ff97942c49c80f3767ed19d891617ae54f9

### CODE REVIEW

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java [hunk 1]
```diff
@@ -273,6 +300,22 @@ public void close() {
 
         this.checkOpened();
         this.sessions.close();
+
+        boolean terminated;
```
Comment (Linary) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java]: `boolean terminated = false` better

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java [hunk 2]
```diff
@@ -273,6 +300,22 @@ public void close() {
 
         this.checkOpened();
         this.sessions.close();
+
```
Comment (javeme) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java]: ensure sessions for all threads are closed

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java [hunk 3]
```diff
@@ -151,18 +161,35 @@ public synchronized void open(HugeConfig config) {
...
+                futures.add(OPEN_POOL.submit(() -> {
+                    this.open(config, disk, disk, Arrays.asList(table));
+                }));
+            }
+        }
+        for (Future<?> f : futures) {
+            try {
+                f.get();
+            } catch (Throwable e) {
+                throw new BackendException("Failed to open RocksDB store", e);
```
Comment (javeme) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java]: seems can destroy thread pool after open

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java [hunk 4]
```diff
@@ -185,13 +185,30 @@ public synchronized void open(HugeConfig config) {
...
-        for (Future<?> f : futures) {
+        for (Future<?> future : futures) {
             try {
-                f.get();
+                future.get();
             } catch (Throwable e) {
                 throw new BackendException("Failed to open RocksDB store", e);
             }
         }
+        this.shutdownOpenPool();
```
Comment (javeme) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java]: should create pool before submit for every open()

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBOptions.java [hunk 5]
```diff
@@ -359,4 +359,28 @@ public static synchronized RocksDBOptions instance() {
...
+                    "rocksdb.max_manifest_file_size",
+                    "The max size of manifest file in bytes.",
+                    rangeInt(1L, Long.MAX_VALUE),
+                    104857600L
+            );
+
+    public static final ConfigOption<Boolean> SKIP_STATS_UPDATE_ON_DB_OPEN =
+            new ConfigOption<>(
+                    "rocksdb.skip_stats_update_on_db_open",
+                    "This flag allows us to not update statistics.",
```
Comment (javeme) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBOptions.java]: Whether to skip statistics update when opening the database, 
setting this flag true allows us to not update statistics.

File: hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java [hunk 6]
```diff
@@ -151,20 +159,59 @@ public synchronized void open(HugeConfig config) {
             return;
         }
 
+        List<Future<?>> futures = new ArrayList<>();
+        ExecutorService openPool = ExecutorUtil.newFixedThreadPool(8, DB_OPEN);
```
Comment (javeme) [inline hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java]: can also use MAX_FILE_OPENING_THREADS?

Comment (zhoney) [reply hugegraph-rocksdb/src/main/java/com/baidu/hugegraph/backend/store/rocksdb/RocksDBStore.java]: this thread pool is used to open DBs, not RocksDB sst files.

**Pull request merged**